### PR TITLE
Feature 2949 cxx11 doc

### LINKED ISSUE #2949: Documentation: Provide instructions for compiling MET with the C++11 standard
## Describe the Task ##
Due to the issue [Modify configure.ac to require the C++17 standard #2948](https://github.com/dtcenter/MET/issues/2948), provide instructions for users can compile MET with the C++11 standard. 

Update the screenshot in this [section](https://met.readthedocs.io/en/develop/Users_Guide/installation.html#executing-the-compile-met-all-sh-script) to include the newer versions of ecKit and ATLAS.

Review the rest of the page for any other necessary changes.

### Time Estimate ###
1-3 days of work

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
MET-12.0.0-rc1 release

### Funding Source ###
TBD

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Review default **alert** labels
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Milestone and Projects ###
- [x] Select **Milestone** as a **MET-X.Y.Z** version, **Consider for Next Release**, or **Backlog of Development Ideas**
- [x] For a **MET-X.Y.Z** version, select the **MET-X.Y.Z Development** project

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the next official version
Select: **MET-X.Y.Z Development** project for development toward the next official release
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by jprestop**
## Expected Differences ##

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **[Not to the tools, only to the compilation configuration]**</br>
If **yes**, please describe:</br>

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Reviewed the changes in RTD on the [Installation page of my feature branch](https://met.readthedocs.io/en/feature_2949_cxx11_doc/Users_Guide/installation.html#installation).

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Review the changes and [formatting](https://met.readthedocs.io/en/feature_2949_cxx11_doc/Users_Guide/installation.html#installation), and suggest any modifications.  Ensure all tests pass.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the MET test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Will this PR result in changes to existing METplus Use Cases? **[No]**</br>
If **yes**, create a new **Update Truth** [METplus issue](https://github.com/dtcenter/METplus/issues/new/choose) to describe them.

- [x] Do these changes introduce new SonarQube findings? **[No]**</br>
If **yes**, please describe:

- [x] Please complete this pull request review by **[20240920]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **MET-X.Y.Z Development** project for official releases
- [x] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Users_Guide/installation.rst [hunk 1]
```diff
@@ -162,18 +162,34 @@ Environment Variables to Run Script
...
+Environment Variable Descriptions section below. An easy way to set these
+environment variables is in an environment configuration file
+(for example, **install_met_env.<machine_name>**). An example environment
+configuration file to start with (**install_met_env.generic_gnu**),
+as well as the environment configuration files used on HPCs at NCAR and NOAA,
 can be found in the `MET GitHub repository <https://github.com/dtcenter/MET>`_ in the 
 `scripts/installation/config <https://github.com/dtcenter/MET/tree/main_v12.0/internal/scripts/installation/config>`_
 directory.
 
+.. note:: Starting with MET-12.0.0, the C++17 standard is the default C++ standard for MET.
```
Comment (JohnHalleyGotway) [inline docs/Users_Guide/installation.rst]:
```suggestion
.. note:: Starting with MET-12.0.0, C++17 is the default C++ standard for MET due to the requirements of its dependent libraries. However, MET itself only makes use of C++11 features.
```

**Review by JohnHalleyGotway — DISMISSED** [review]
Julie, I already pushed some whitespace changes and proposed one small change.

I do note that `12.0` appears 20 times in here, and I'm always worried about preventing the docs from looking stale. Should we instead define an env var at the beginning in one spot and then reference it throughout? As the documentation lead, its up to you.

Could be something like
```
export MET_VERSION=12.0.0
```
And then reference `${MET_VERSION}` in subsequent instructions.

**Comment by jprestop** [comment]
> Julie, I already pushed some whitespace changes and proposed one small change.
> 
> I do note that `12.0` appears 20 times in here, and I'm always worried about preventing the docs from looking stale. Should we instead define an env var at the beginning in one spot and then reference it throughout? As the documentation lead, its up to you.
> 
> Could be something like
> 
> ```
> export MET_VERSION=12.0.0
> ```
> 
> And then reference `${MET_VERSION}` in subsequent instructions.

@JohnHalleyGotway I accepted your suggested small change.  Thank you for that suggestion!

I agree and had the same concerns.  I had talked with @j-opatz and @TaraJensen and they both agreed that the specific version number added clarity and ease for the user.   I did update the release notes (and assigned that PR to you also, but I don't think you've had a chance to review it yet) which adds the following under Update Version Number:

> In *docs/Users_Guide/installation.rst*, search for the X.Y version, replacing the current X.Y version with the official X.Y version, if necessary. Pay particular attention to the "Note" about the C++ standard and modify if necessary. The X.Y version number in the "Note" box should NOT change unless the default C++ standard changes.

Let me revisit this and see if I can add a new variable in conf.py for just the X.Y instead of the X.Y.Z with possible "-betaN" or -rcN" because sometimes we need just X.Y.  For example:

> wget https://raw.githubusercontent.com/dtcenter/MET/main_v12.0/internal/scripts/installation/compile_MET_all.sh

> `compile_MET_all.sh <https://raw.githubusercontent.com/dtcenter/MET/main_v12.0/internal/scripts/installation/compile_MET_all.sh>`_

but, I'm not sure I can successfully replace all of those instances, particularly the links.  If not, I can at least reduce the instances of 12.0.  I'll follow up once I test some things out.  Unless you think that the instructions I added to the release guide are sufficient.  Please let me know your thoughts when you get a chance.

**Review by JohnHalleyGotway — DISMISSED** [review]
I approve.

Julie, you're aware of the version-itis challenge, thought through the options, and came to a conclusion. That works for me! Thanks for thinking through the details.

**Comment by jprestop** [comment]
Thanks @JohnHalleyGotway.  Unfortunately, I didn't have the file back in place after testing.  When I put it back up, GitHub dismissed your stal review:

> jprestop dismissed JohnHalleyGotway’s stale review via 52a0ef4 17 minutes ago

so I need to re-request your review.

**Pull request merged**